### PR TITLE
Update itf

### PR DESCRIPTION
**Pull request by kamilsa**
Signed-off-by: kamilsa <kamilsa16@gmail.com>

### Description of the Change

Adds shared model to itf

### Benefits

ITF has shared model

### Possible Drawbacks 

None

### CODE REVIEW

**Comment by l4l** [comment]
Quite misleading names (pr/commit), it changes iroha instance iface, not itf

File: test/framework/integration_framework/iroha_instance.cpp [hunk 1]
```diff
@@ -34,19 +34,19 @@ namespace integration_framework {
         vote_delay_(5000ms),
```
Comment (victordrobny) [inline test/framework/integration_framework/iroha_instance.cpp]: Line 20: include is not necessary

File: test/framework/integration_framework/iroha_instance.hpp [hunk 2]
```diff
@@ -24,6 +24,7 @@
 #include "common/types.hpp"
```
Comment (victordrobny) [inline test/framework/integration_framework/iroha_instance.hpp]: It seems to be unnecessary

File: test/framework/integration_framework/iroha_instance.hpp [hunk 3]
```diff
@@ -24,6 +24,7 @@
...
 #include "framework/config_helper.hpp"
 #include "integration/pipeline/test_irohad.hpp"
+#include "cryptography/keypair.hpp"
 
 namespace iroha {
   namespace model {
```
Comment (victordrobny) [inline test/framework/integration_framework/iroha_instance.hpp]: remove forward declaration of old model

File: test/integration/pipeline/test_irohad.hpp [hunk 4]
```diff
@@ -42,7 +43,7 @@ class TestIrohad : public Irohad {
                proposal_delay,
                vote_delay,
                load_delay,
-               keypair) {}
+               *std::shared_ptr<iroha::keypair_t>(keypair.makeOldModel())) {}
```
Comment (victordrobny) [inline test/integration/pipeline/test_irohad.hpp]: maybe unique_ptr is more convenient here?

Comment (victordrobny) [reply test/integration/pipeline/test_irohad.hpp]: However it does not matter

**Pull request merged**